tests/integration-tests/tests/scaling/test_scaling.py

```diff
@@ -69,31 +72,209 @@ def test_multiple_jobs_submission(scheduler, region, pcluster_config_reader, clu
+    instance_ids_to_hostname = get_instance_ids_to_compute_hostnames_dict(instance_ids)
+
+    logging.info("Testing that nodewatcher will terminate a node in failing state")
+    # submit a job to run on all nodes
+    scheduler_commands.submit_command("sleep infinity", nodes=initial_queue_size)
+    expected_num_nodes_killed = 4 if scheduler == "slurm" else 1
+    # simulate unexpected hardware failure by killing first x nodes
+    nodes_to_remove = compute_nodes[:expected_num_nodes_killed]
+    for node in nodes_to_remove:
+        remote_command_executor.run_remote_script(
+            str(test_datadir / "{0}_kill_scheduler_job.sh".format(scheduler)), args=[node]
+        )
+    failed_instances = wait_compute_log(remote_command_executor, expected_num_nodes_killed)
 
-    # submit a job that kills the slurm daemon so that the node enters a failing state
-    scheduler_commands.submit_script(str(test_datadir / "{0}_kill_scheduler_job.sh".format(scheduler)))
-    instance_id = wait_compute_log(remote_command_executor)
+    for instance_id in failed_instances:
+        assert_that(nodes_to_remove).contains(instance_ids_to_hostname.get(instance_id))
+        _assert_compute_logs(remote_command_executor, instance_id)
+        assert_instance_replaced_or_terminating(instance_id, region)
 
-    _assert_compute_logs(remote_command_executor, instance_id)
-    assert_instance_replaced_or_terminating(instance_id, region)
-    # verify that desired capacity is still 1
-    assert_that(get_desired_asg_capacity(region, cluster.cfn_name)).is_equal_to(1)
-    _assert_nodes_removed_from_scheduler(scheduler_commands, compute_nodes)
+    nodes_to_retain = [compute for compute in compute_nodes if compute not in nodes_to_remove]
+    # verify that desired capacity is still the initial_queue_size
+    assert_that(get_desired_asg_capacity(region, cluster.cfn_name)).is_equal_to(initial_queue_size)
+    _assert_nodes_removed_and_replaced_in_scheduler(
+        scheduler_commands, nodes_to_remove, nodes_to_retain, desired_capacity=initial_queue_size
+    )
 
     assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
 
 
-@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+@pytest.mark.regions(["us-west-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["slurm"])
+@pytest.mark.os(["ubuntu1804"])
+@pytest.mark.usefixtures("region", "os", "instance")
+@pytest.mark.scaling_with_manual_actions
+def test_scaling_with_manual_actions(scheduler, region, pcluster_config_reader, clusters_factory):
+    """Test that slurm-specific scaling logic is resistent to manual actions and failures."""
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    num_compute_nodes = 5
```

Comment:
I dropped a comment in corresponding config file about why I think it'd be a good idea to pass this to pcluster_config_reader.